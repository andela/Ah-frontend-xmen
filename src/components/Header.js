import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <div>
    <NavLink to="/" exact>Home</NavLink>
     &nbsp;
    <NavLink to="/login" exact>Login</NavLink>
    <NavLink to="/editor" exact>Create an Article</NavLink>
    <NavLink to="/editor/:slug" exact>Edit an Article</NavLink>
  </div>
);

export default Header;
