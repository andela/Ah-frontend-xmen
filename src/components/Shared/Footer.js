import React from 'react';

const Footer = () => (

  <footer className="footer shadow">
    <img src="https://res.cloudinary.com/soultech/image/upload/v1550685426/authors%20haven%20pics/logo4authorsHaven.png" alt="" />
    <p className="copyright">
      &copy; Authors Haven 2019
      <br />
      Project by X-men
      <br />
      <small className="text-primary">Andela Uganda</small>
    </p>
  </footer>
);

export default Footer;
