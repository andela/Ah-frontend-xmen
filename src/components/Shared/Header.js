import React, { Component } from 'react';
import { Link, Redirect } from 'react-router-dom';
import RegisterModal from '../auth/RegisterModal';

class Header extends Component {
  state = {
    open: false,
  }

  onModalOpen = (event) => {
    event.preventDefault();
    this.setState({ open: true });
  }

  onModalClose = (event) => {
    event.preventDefault();
    this.setState({ open: false });
    return <Redirect to="/" />;
  }

  render() {
    const { open } = this.state;
    return (
      <div>
        <nav className="navbar navbar-expand-lg shadow navbar-dark">
          <div className="container">
            <Link to="/"><img src="https://res.cloudinary.com/soultech/image/upload/v1550685433/authors%20haven%20pics/logo4authorsHaven1.png" alt="" /></Link>
            <button type="submit" id="modalOpener" onClick={this.onModalOpen} className="btn login-nav button-primary">Get Started</button>
          </div>
        </nav>
        <RegisterModal open={open} onModalClose={this.onModalClose} />
      </div>
    );
  }
}

export default Header;
