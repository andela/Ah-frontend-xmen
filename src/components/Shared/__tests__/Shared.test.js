import React from 'react';
import { shallow } from 'enzyme';
import { Redirect } from 'react-router-dom';
import Footer from '../Footer';
import { Header } from '../Header';

const props = {
  getAuthUserDetails: jest.fn(),
  auth: {
    IsAuth: true,
  },
};

describe('Shared Components', () => {
  it('should render Footer without crushing', () => {
    const component = shallow(<Footer />);
    expect(component).toMatchSnapshot();
  });

  it('should render Header without crushing', () => {
    const component = shallow(<Header {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should call onModalClose', () => {
    const wrapper = shallow(<Header {...props} />);
    expect(wrapper.instance().onModalClose({ preventDefault() {} })).toEqual(<Redirect push={false} to="/" />);
  });

  it('should call onModalOen', () => {
    const wrapper = shallow(<Header {...props} />);
    const event = { preventDefault: jest.fn() };
    wrapper.instance().onModalOpen(event);
    expect(event.preventDefault).toHaveBeenCalled();
  });

  it('should open dropdown on click', () => {
    const wrapper = shallow(<Header {...props} />);
    wrapper.instance().setState = jest.fn();
    wrapper.find('#dd-click').simulate('click', { preventDefault: () => {} });
    expect(wrapper.instance().setState).toHaveBeenCalled();
  });

  it('should close dropdown on click', () => {
    const wrapper = shallow(<Header {...props} />);
    wrapper.instance().setState = jest.fn();
    wrapper.instance().state.isShowing = true;
    wrapper.find('#dd-click').simulate('click', { preventDefault: () => {} });
    expect(wrapper.instance().setState).toHaveBeenCalled();
  });

  it('should log out', () => {
    const wrapper = shallow(<Header {...props} />);
    localStorage.setItem('slug', 'it-is-me');
    wrapper.instance().logout = jest.fn();
    wrapper.find('#logout').simulate('click', { preventDefault: () => {} });
    expect(localStorage.getItem('slug')).toBeNull();
  });
});
