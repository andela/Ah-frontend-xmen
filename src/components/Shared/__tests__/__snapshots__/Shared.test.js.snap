// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shared Components should render Footer without crushing 1`] = `ShallowWrapper {}`;

exports[`Shared Components should render Header without crushing 1`] = `ShallowWrapper {}`;
